### PR TITLE
PMM-13145 Certs Fix - Backward compatibility for older clients. (#3014).

### PR DESCRIPTION
**Pull request by JiriCtvrtka**
* PMM-13145 Fix for older clients.

* PMM-13145 Small changes.

* PMM-13145 Lint.

* PMM-13145 Lint.

* PMM-13145 Logic fix.

* PMM-13145 Refactor.

* PMM-13145 Fix after refactor.

* Update managed/models/agent_model.go

* Update managed/models/agent_model.go

---------

PMM-0

Link to the Feature Build: SUBMODULES-0

If this PR adds or removes or alters one or more API endpoints, please review and add or update the relevant [API documents](https://github.com/percona/pmm/tree/main/docs/api) as well:

- [ ] API Docs updated

If this PR is related to some other PRs in this or other repositories, please provide links to those PRs:

- Links to related pull requests (optional).

### CODE REVIEW

**Comment by JiriCtvrtka** [comment]
I dont have permissions to merge into this branch at all, please merge it once it will get enough approvals.

**Review by ademidoff — APPROVED** [review]

**Pull request merged**
